### ISSUE TITLE
INV-011: Auto-Discovery Integration (GitHub / AWS / Azure / GCP)

### ISSUE THREAD

**Issue by Akshay-Exelixi** [Akshay-Exelixi/AI-GRC#12]
## Story: INV-011 — Auto-Discovery Integration (GitHub / Cloud Platforms)

**EPIC:** #1 Centralized AI Inventory  
**Priority:** 🟡 P1-SHOULD (Deferred for Demo — P2 in Sprint Plan)  
**Story Points:** 8  
**Sprint:** Post-Sprint 4 (Backlog)  

---

### User Story
> As a **SISO**, I want the platform to automatically discover AI systems deployed in our GitHub repositories and cloud platforms (AWS, Azure, GCP), so that I can identify shadow AI and ensure nothing is missed in the inventory.

### Description
Implement automated AI discovery connectors (FR-003) that scan code repositories and cloud environments to identify AI/ML systems not yet registered in the inventory.

### Discovery Sources

**GitHub / GitLab:**
- Scan repositories for ML/AI indicators:
  - `requirements.txt` / `pyproject.toml` with: tensorflow, pytorch, scikit-learn, transformers, langchain, openai, anthropic
  - Dockerfile with ML framework base images
  - `.github/workflows/` with ML pipeline steps
  - Jupyter notebooks (`.ipynb` files)
  - Model files (`.pkl`, `.h5`, `.onnx`, `.pt`, `.safetensors`)
- Extract: repo name, description, last commit, contributors, dependencies

**AWS:**
- Scan for: SageMaker endpoints, Bedrock models, Lambda functions with AI SDK, Rekognition, Comprehend, Textract
- Extract: service name, region, tags, status, last invocation

**Azure:**
- Scan for: Azure ML endpoints, Cognitive Services, OpenAI Service deployments
- Extract: resource name, resource group, status, tags

**GCP:**
- Scan for: Vertex AI endpoints, AutoML models, Cloud Functions with AI libraries
- Extract: project, model name, status, labels

### Data Flow
```
Discovery Source → Connector → Scan → discovery_results table → 
  Status: New → Review by SISO → Register (link to ai_systems) or Dismiss
```

### Tables
- `discovery_sources`: Connection configs per source (encrypted)
- `discovery_results`: Each discovered item with status tracking

### Acceptance Criteria
- [ ] GitHub connector discovers repos with ML dependencies
- [ ] At least 1 cloud connector (AWS or Azure) discovers AI services
- [ ] Discovery results appear in dedicated "Discovery" page
- [ ] Each result shows: source, name, type, URL, metadata
- [ ] SISO can review → Register (creates AI system) or Dismiss
- [ ] Registered results link to the created AI system
- [ ] Scan runs on schedule (daily/weekly) or manually
- [ ] Connection config stored encrypted
- [ ] Duplicate detection: if already registered, flag as "Already Tracked"

### Definition of Done
- [ ] Code reviewed and merged
- [ ] At least GitHub connector fully working
- [ ] Mock data available for cloud connector demo
- [ ] API documented

### Notes
> This story is **deferred** from the 8-week sprint plan. Manual + CSV import covers the demo. Build post-April 19.
